db/schema.rb

```diff
@@ -40,6 +40,15 @@
     t.index ["sub"], name: "index_oidc_users_on_sub", unique: true
   end
 
+  create_table "saved_pages", force: :cascade do |t|
+    t.bigint "oidc_user_id", null: false
+    t.string "page_path", null: false
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["oidc_user_id", "page_path"], name: "index_saved_pages_on_oidc_user_id_and_page_path", unique: true
+    t.index ["oidc_user_id"], name: "index_saved_pages_on_oidc_user_id"
```

Comment:
This looks good, do you think
`t.index ["page_path"], name: "index_saved_pages_on_page_path"`
would also be needed?

Reply:
I don't _think_ so... let me step through this and check on the why.

For the index path we will look up saved pages by `oicd_user_id` and then return all. So in that case there's no page path, but we may get a nice speed bump from the index.

However there is never a case where we will be looking up a page path independent of a user. It is always an authenticated user looking up either all their saved pages or making a request where we use both their ID and a page path?

I guess there could be a scenario where we start reporting on pages saved and want to look them up, but in that case it could be a rake task, where perhaps speed is not the issue? Or we could add an index then in response to a need?

Reply:
Sounds good, we'll cross that bridge when we get to it.